### PR TITLE
Git step documentation update

### PR DESCRIPTION
**Pull request by sa2ajj**
## Contributor Checklist:

* [x] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation

----

I finally found an internal project to use the latest Buildbot and while the only important change is about being able to use tags, I decided to polish the step's documentation.

Requests for more documentation updates are more than welcome :smile:

### CODE REVIEW

File: master/buildbot/newsfragments/git-source-step-update.doc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Update Git source checkout build step documentation:
+
+* remove redundant empty lines
```
Comment (tardyp) [inline master/buildbot/newsfragments/git-source-step-update.doc]: I guess the newsfragment should be reasonably concise :)

Comment (sa2ajj) [reply master/buildbot/newsfragments/git-source-step-update.doc]: would you like me to make it one liner?  (do we have a policy about this somewhere?)

Comment (tardyp) [reply master/buildbot/newsfragments/git-source-step-update.doc]: There is a readme file in this directory, which explains the policy. But yes, this is what is made into the changelog. So it should be concise and contain information useful for a changelog.

**Comment by sa2ajj** [comment]
I'm sorry, I did not provide a fix for the news item :(

**Comment by tardyp** [comment]
no problem :) I often fix news fragment. Github makes it very easy

**Pull request merged**